pkg/policy/policy.go

```diff
@@ -26,6 +27,18 @@ import (
 	"github.com/op/go-logging"
 )
 
+// Completion is an interface for an individual asynchronous success/failure
+// notification.
+type Completion interface {
```

Comment:
This could simply be defined as a func:
```
type Completion func(success bool)
```

Reply:
An interface carries an object reference with it, while a function pointer does not. Multiple different endpoints may request completion callbacks in parallel from the same stream (in case of LDS).